### PR TITLE
platform: document default installation files

### PR DESCRIPTION
**Pull request by rubensfig**
Document bisdn/meta-switch/pull/203 functionality to provide default installation of l2 functionality.

Addresses https://github.com/bisdn/bisdn-linux/issues/101.

### CODE REVIEW

File: platform_configuration/default_installer.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+---
+
+### Default Installer Files
+
+**WARNING** This page provides a tool to provide default configuration in your switch. Only run this command if you are prepared to lose your configuration!
+
+The `baseboxd` package includes default systemd-networkd configuration files. These files initialize bisdn-linux to provide a default l2 switch forwarding functionality. To configure this, pass the `BISDN_DEFAULT_CONFIG` environment variable in the ONIE environment.
+
+```
+BISDN_DEFAULT_CONFIG="simple-l2-bridge" onie-nos-install http://repo.bisdn.de.s3-eu-central-1.amazonaws.com/pub/onie/generic-x86-64/onie-bisdn-generic-x86-64-v5.1.1.bin
```
Comment (ideaship) [inline platform_configuration/default_installer.md:14]: You can shorten the hostname to repo.bisdn.de.
Also, given that 5.1.1 does not have this functionality, maybe replace v5.1.1 with [...] or something.

File: platform_configuration/default_installer.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+title: Default Installer Files
+parent: Platform Configuration
+nav_order: 3
+---
+
+### Default Installer Files
+
+**WARNING** This page provides a tool to provide default configuration in your switch. Only run this command if you are prepared to lose your configuration!
```
Comment (ideaship) [inline platform_configuration/default_installer.md:9]:
```suggestion
**WARNING** This page documents a mechanism for reinstalling your switch with a default configuration. Only run this command if you are prepared to lose all files in your current BISDN Linux installation.
```
Just a suggestion, but we to make clear that the user is not only losing /etc/systemd/network or other configuration files, but _all_ files. The mechanism ignores system-backup.txt and user-backup.txt. It is more like onie-bisdn-uninstall than onie-bisdn-upgrade.

File: platform_configuration/default_installer.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+title: Default Installer Files
+parent: Platform Configuration
+nav_order: 3
+---
+
+### Default Installer Files
+
+**WARNING** This page documents a mechanism for reinstalling your switch with a default configuration. Only run this command if you are prepared to lose all files in your current BISDN Linux installation.
+
+The `baseboxd` package includes default systemd-networkd configuration files. These files initialize bisdn-linux to provide a default l2 switch forwarding functionality. To configure this, pass the `BISDN_DEFAULT_CONFIG` environment variable in the ONIE environment.
```
Comment (KanjiMonster) [inline platform_configuration/default_installer.md:11]: That these configuration files come from the baseboxd package is an implementation detail that really doesn't matter for users.

I think this paragraph currently could improve the explanation for why a user would want to use it. E.g. something like

```suggestion
BISDN Linux by default provides no network configuration for the switch ports. To allow a more switch like default configuration, it is possible to use the environment variable `BISDN_DEFAULT_CONFIG` while installing BISDN Linux to provide a default network configuration.
```

File: platform_configuration/default_installer.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+title: Default Installer Files
+parent: Platform Configuration
+nav_order: 3
+---
+
+### Default Installer Files
+
+**WARNING** This page documents a mechanism for reinstalling your switch with a default configuration. Only run this command if you are prepared to lose all files in your current BISDN Linux installation.
```
Comment (KanjiMonster) [inline platform_configuration/default_installer.md:9]: The wording is a bit redundant, no need to reference that this is a page.

```suggestion
**WARNING** Setting a default configuration will reset the file system to the initial state. Only use this if you are prepared to lose all modifications of your current BISDN Linux installation.
```

**Review by KanjiMonster — CHANGES_REQUESTED** [review]
I think it would make more sense to put this into https://github.com/bisdn/bisdn.github.io/blob/main/getting_started/install_bisdn_linux.md instead of a separate page, since you might want to know about this before doing an installation.

**Review by rubensfig — COMMENTED** [review]
Addressed all comments

File: getting_started/install_bisdn_linux.md [hunk 5]
```diff
@@ -65,6 +65,18 @@ onie-nos-install http://repo.bisdn.de/pub/onie/agema-ag7648/onie-bisdn-agema-ag7
...
+
+**WARNING** Setting a default configuration will reset the file system to the initial state. Only use this if you are prepared to lose all modifications of your current BISDN Linux installation.
+
+BISDN Linux by default provides no network configuration for the switch ports. To allow a more switch like default configuration, it is possible to use the environment variable `BISDN_DEFAULT_CONFIG` while installing BISDN Linux to provide a default network configuration.
+
+```
+BISDN_DEFAULT_CONFIG="simple-l2-bridge" onie-nos-install http://repo.bisdn.de/pub/onie/generic-x86-64/onie-bisdn-generic-x86-64-v5.1.X.bin
+```
+
+The `baseboxd` supported files are in the file system under the `/usr/share/baseboxd/default_configurations/` path.
```
Comment (KanjiMonster) [inline getting_started/install_bisdn_linux.md:78]: I don't think this helps and just confuses (from a user perspective what has this to do with baseboxd? how would you access this path?).

What would be nice to have a table here with all supported values and an explanation of it.

* (empty) => just takes over the previous installation's configuration, if there is one (default)
* simple-l2-bridge => removes existing configuration, puts all ports into a bridge, behaves like an unmanaged switch
* none => removes existing configuration, resetting the installation to the default state with no configuration on switch ports

File: getting_started/install_bisdn_linux.md [hunk 6]
```diff
@@ -65,6 +65,23 @@ onie-nos-install http://repo.bisdn.de/pub/onie/agema-ag7648/onie-bisdn-agema-ag7
...
+```
+BISDN_DEFAULT_CONFIG="simple-l2-bridge" onie-nos-install http://repo.bisdn.de/pub/onie/generic-x86-64/onie-bisdn-generic-x86-64-v5.1.X.bin
+```
+
+The following table shows the supported configuration for `BISDN_DEFAULT_CONFIG` and the expected outcome.
+
+|`BISDN_DEFAULT_CONFIG` Value|Outcome|Default|
+|(Empty)|Use previous installation configuration|Y|
+|none|Removes existing configuration and resets the switch to no default state||
+|simple-l2-bridge|Removes existing configuration and initializes the switch to behave as an unmanaged switch||
```
Comment (KanjiMonster) [inline getting_started/install_bisdn_linux.md:83]: 1. Lets align the table in the source as well, to make it more readable on the command line.
2. Lets be more explicit in explaining the network configuration options.

```suggestion
|`BISDN_DEFAULT_CONFIG` Value|Outcome                                                                                        |Default|
|(Empty)                     |Use previous installation configuration                                                        |Y      |
|`none`                      |Removes existing configuration and initializes the switch with all ports disabled              |       |
|`simple-l2-bridge`          |Removes existing configuration and initializes the switch to forward traffic between all ports |       |
```

**Review by KanjiMonster — CHANGES_REQUESTED** [review]

**Review by KanjiMonster — APPROVED** [review]
LGTM

**Pull request merged**